knowledge_graphs/query_pagination.py

```
@@ -0,0 +1,382 @@
+            
+        Returns:
+            Optimized query
+        """
+        # Basic optimizations
+        optimized = query
+        
+        # Add hints for common patterns
+        if 'MATCH (r:Repository)' in query and 'r.name' in query:
+            optimized = optimized.replace(
+                'MATCH (r:Repository)',
+                'MATCH (r:Repository) USING INDEX r:Repository(name)'
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)